gcode/builder.go

```diff
@@ -151,17 +157,36 @@ func (g *Builder) AddPolygon(currentLayer data.PartitionedLayer, polygon data.Pa
 			}
 
 			if isCrossing {
```

Comment:
Please make sure that if zHop is 0, the resulting gcode should not differ from the gcode generated before your changes.

-> Please use the g.AddCommand like before to preserve the special retraction speed. That retraction speed should also apply if zHop is enabled. 
-> make sure that the extra moves for the zHop are only added when zHop is enabled. (> 0)